aws/rust-runtime/aws-http/src/retry.rs

```diff
@@ -39,41 +28,38 @@ const THROTTLING_ERRORS: &[&str] = &[
 ];
 const TRANSIENT_ERRORS: &[&str] = &["RequestTimeout", "RequestTimeoutException"];
 
-impl AwsErrorRetryPolicy {
-    /// Create an `AwsErrorRetryPolicy` with the default set of known error & status codes
+/// Implementation of [`ClassifyResponse`] that models AWS error codes.
+///
+/// In order of priority:
+/// 1. The `x-amz-retry-after` header is checked
+/// 2. The modeled error retry mode is checked
+/// 3. The code is checked against a predetermined list of throttling errors & transient error codes
+/// 4. The status code is checked against a predetermined list of status codes
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub struct AwsResponseClassifier;
+
+impl AwsResponseClassifier {
+    /// Create an `AwsResponseClassifier` with the default set of known error & status codes
     pub fn new() -> Self {
-        AwsErrorRetryPolicy
+        AwsResponseClassifier
     }
 }
 
-impl Default for AwsErrorRetryPolicy {
+impl Default for AwsResponseClassifier {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl<T, E> ClassifyResponse<T, SdkError<E>> for AwsErrorRetryPolicy
+impl<T, E> ClassifyResponse<T, SdkError<E>> for AwsResponseClassifier
 where
     E: ProvideErrorKind,
 {
-    fn classify(&self, err: Result<&T, &SdkError<E>>) -> RetryKind {
-        let (err, response) = match err {
-            Ok(_) => return RetryKind::Unnecessary,
-            Err(SdkError::ServiceError { err, raw }) => (err, raw),
-            Err(SdkError::TimeoutError(_err)) => {
-                return RetryKind::Error(ErrorKind::TransientError)
-            }
-
-            Err(SdkError::DispatchFailure(err)) => {
-                return if err.is_timeout() || err.is_io() {
-                    RetryKind::Error(ErrorKind::TransientError)
-                } else if let Some(ek) = err.is_other() {
-                    RetryKind::Error(ek)
-                } else {
-                    RetryKind::UnretryableFailure
-                };
-            }
-            Err(_) => return RetryKind::UnretryableFailure,
+    fn classify(&self, result: Result<&T, &SdkError<E>>) -> RetryKind {
+        let (err, response) = match DefaultResponseClassifier::try_extract_err_response(result) {
```

Comment:
Could you add a comment here and on line 63 about how this is the basic retry classification and how the AWS-specific retry logic is the fallback? I was a bit confused as to why all logic wasn't encompassed by the `try_extract_err_response` function.